### PR TITLE
feat(perf-detector-threshold-configuration) Added frontend for N+1 and Slow db queries.

### PR DESCRIPTION
**Pull request by Abdkhan14**
- Added frontend for project performance settings: 
Screenshot 2023-06-21 at 1 11 48 PM

- Added tests. 
- Removed some admin ui options for now, will add back in the coming PRs.

### CODE REVIEW

File: static/app/views/settings/projectPerformance/projectPerformance.tsx [hunk 1]
```diff
@@ -118,6 +143,23 @@ class ProjectPerformance extends AsyncView<Props, State> {
...
+      loading: true,
+    });
+
+    this.api.request(
+      `/projects/${organization.slug}/${projectId}/performance-issues/configure/`,
+      {
+        method: 'DELETE',
+        complete: () => this.fetchData(),
+      }
+    );
```
Comment (narsaynorath) [inline static/app/views/settings/projectPerformance/projectPerformance.tsx:160]: We talked about this but I just wanted to comment here that I noticed this doesn't just reset the thresholds panel but all of the performance issue stuff. Including the issue occurrences flags and the admin options. But we agreed we could improve this in a new PR

**Review by narsaynorath — APPROVED** [review]
A few comments but mostly looks good to me. The threshold reset issue can be handled separately since this is still feature flagged

File: static/app/components/forms/formPanel.tsx [hunk 2]
```diff
@@ -69,39 +69,37 @@ function FormPanel({
           )}
         </PanelHeader>
       )}
-      {!collapsed && (
-        <PanelBody>
-          {typeof renderHeader === 'function' && renderHeader({title, fields})}
+      <StyledPanelBody collapsed={collapsed}>
```
[the remark this answers is not in the document]
Comment (Abdkhan14) [reply static/app/components/forms/formPanel.tsx]: This is way better thanks.

File: static/app/views/settings/projectPerformance/projectPerformance.tsx (hunk 1 above)
Comment (Abdkhan14) [reply static/app/views/settings/projectPerformance/projectPerformance.tsx]: I added the reset as a put request for now. Also, added a todo [here](https://www.notion.so/sentry/Detector-Threshold-Configuration-f8cb07e7cceb42388cedb09ea05fc116?pvs=4#98f4af6b47f44ab295864886ec870976).

**Pull request merged**